assets/js/utils/rtcUtils.ts

```diff
@@ -178,80 +198,67 @@ export const connectWebrtc = async (
   return webrtc;
 };
 
-export const connectPresentersTracks = async (
-  playerCallbacks: { [key: string]: (sourceType: SourceType) => void },
-  setPresenters: React.Dispatch<React.SetStateAction<{ [key: string]: Presenter }>>
-) => {
-  setPresenters((presenters) => {
-    Object.values(presenters)
-      .filter(
-        (presenter) =>
-          presenter.rtcStatus == "rtc_player_ready" && presenter.connectCallbacks.length > 0
-      )
-      .forEach((presenter) => {
-        const playerCallback = playerCallbacks[presenter.email];
-        presenter.connectCallbacks.forEach((callback) => callback(playerCallback));
-        presenters = { ...presenters, [presenter.email]: { ...presenter, connectCallbacks: [] } };
-      });
-    return presenters;
-  });
-};
-
 export const changeSource = async (
   webrtc: MembraneWebRTC | null,
   client: Client,
   deviceId: string,
   sourceType: SourceType,
-  playerCallback: (sourceType: SourceType) => void
+  presenterStreams: PresenterStreams,
+  setPresenterStreams: React.Dispatch<React.SetStateAction<PresenterStreams>>
 ) => {
-  await setSourceById(client, deviceId, sourceType, playerCallback);
+  await setSourceById(client, deviceId, sourceType, setPresenterStreams);
   if (!webrtc) return;
   if (mergedScreenRef.refreshId && sourceType == "video") {
-    shareScreen(webrtc, client, playerCallback);
+    shareScreen(webrtc, client, presenterStreams, setPresenterStreams);
   } else {
```

Comment:
This way we could remove one nesting level.
```suggestion
  if (mergedScreenRef.refreshId && sourceType == "video") {
    shareScreen(webrtc, client, playerCallback);
    shareScreen(webrtc, client, presenterStreams, setPresenterStreams);
    return;
  } 
```